src/SegmentingContainer.php

```diff
@@ -7,42 +7,47 @@
 use function ltrim;
 
 /**
- * A container implementation that wraps around another to allow the container to be segmented into sub-containers.
+ * This container implementation decorates another to provide nested container access even when the decorated
+ * container's internal data is flat.
  *
- * Segmenting containers are intended to be used with hierarchical keys, i.e. keys that use a delimiter to indicate
- * segments in the key, such as "some/test/key" or "some.deep.config.value".
+ * Segmenting containers are intended to be used with keys that contain segments, i.e. keys that use a delimiter to
+ * indicate hierarchy. For example: "some/test/key" or "some.deep.config.value". The delimiter can be configured during
+ * construction of a segmenting container.
  *
- * A segmenting container can yield 2 different kinds of results:
- * * values
- * * "segments"
+ * A segmenting container can yield 2 different kinds of results when {@link SegmentingContainer::get()} is called:
  *
- * When `get()` is called, the key is split into segments according to the delimiter, and those segments are appended
- * to an internal segment list (more on this further below) to obtain the full key.
+ * **Values**
  *
- * If the full key corresponds to a value in the inner container, the value for that key is returned.
+ * If the inner container has a value for the given key, that value is returned.
  *
- * Otherwise, a new segmenting container is created and returned. This segmenting container will store the full key
- * that resulted in its creation as an internal segment list, which will be automatically prepended to all keys that
- * are requested through its `get()` method.
+ * **Segments**
  *
- * Example usage:
- *      Consider the hierarchy:
- *          [
- *              "config" => [
- *                  "db" => [
- *                      "host" => "localhost",
- *                      "port" => 3306
- *                  ]
- *              ]
- *          ]
+ * If the inner container has no value for the given key, a new {@link SegmentingContainer} instance is returned. This
+ * segmenting container will be aware of the key that resulted in its creation, and will automatically prepend that key
+ * to parameter keys given in `get()`.
  *
- *      The segmenting container can create a container that directly provides the "host" and "port":
- *          $config = new SegmentingContainer($c, '.');
- *          $dbConfig = $config->get('config.db');
- *          $dbConfig->get("host"); // "localhost"
- *          $dbConfig->get("port"); // 3306
+ * **Example usage:**
+ *
+ * Consider the below data and a regular `$container` that provides access to it:
+ *
+ * ```php
+ * $data = [
```

Comment:
Forgot to create the container from the data?

Reply:
I omitted it for brevity. See the sentence immediately above it:

> Consider **the below data** and a regular `$container` that provides access to it

For the example, the internal data is more important than the container that wraps around it. The doc asks the reader to consider that data, and to also keep in mind that (obviously) there is a container wrapping around it that provides untempered access to it.

Reply:
The reason I mentioned this is that the next example features a `SegmentingContainer` that accepts a _container_, and so it could be a little confusing as it is now. But maybe not. OK to leave it as is.